src/security/srv_acl.c

```diff
@@ -209,5 +209,9 @@ ds_sec_can_pool_connect(const struct pool_prop_ugm *ugm, d_iov_t *cred,
 	access_permitted = (ugm->pp_mode >> shift) & DAOS_PC_MASK;
 
 	/* Only if all requested capability bits are permitted... */
-	return (access & access_permitted) == access;
+	if ((access & access_permitted) == access) {
+		return 0;
+	} else {
```

Comment:
Should remove the else block and just return -DER_NO_PERM at the end

Reply:
fixed